flink-core/src/main/java/org/apache/flink/api/java/typeutils/runtime/PojoComparator.java

```diff
@@ -176,14 +181,25 @@ public void getFlatComparator(List<TypeComparator> flatComparators) {
 	 */
 	public final Object accessField(Field field, Object object) {
 		try {
-			object = field.get(object);
+			if (field.isAccessible()) {
```

Comment:
wouldn't it be easier (and also more performant) to simply make the field accessible in the constructor of the comparator? Going through all methods for every field access is not very beautiful.

Reply:
It would. I changed it.